### ISSUE TITLE
aria-owns have tabs in wrong order if some tabs have conditional statements

### ISSUE THREAD

**Issue by rbirkgit** [vmware/clarity#5133]
## Describe the accessibility issue

We have a conditional statement on some of our tabs. This causes the aria-owns attribute to list the tabs in the wrong order, putting the tabs with conditions last. This seems to be because tabs put data in an internal array in the order they get registered and that array is then used to populate the aria-owns attribute.

Result is that screen readers then narrates the tab causing the first tab to say 4 of 4, second tab 1 of 4. I.e. strange order.

## WCAG Guideline

WCAG1.3.1

## How to reproduce

Example of our first tab:

```
<clr-tabs>
  <clr-tab *ngIf="!isEmbedded">
    <button type="button" clrTabLink id="summary">Summary</button>

```

Resulting in generated DOM. As you see, the summary tab is put last where it should have been first:

`<ul class="nav" clrdirection="both" role="tablist" aria-owns="subnets isolation securityGroups summary">
`

Reproduced it with Clarity v4:
https://stackblitz.com/edit/clarity-v4-light-theme-tezxvg 

## Expected behavior

Aria attribute to list tabs in the same order as in the html.

## Versions

**App**

- Clarity: 2, 3 and 4

**Device:**

- Browser Chrome 85
- Screen Reader: NVDA

**Comment by gnomeontherun** [comment]
There are two possible workarounds on the app side. The challenge is that the ngIf defers rendering and we are seeing lifecycle issues.

1) Have the whole tabs also with an ngIf to have it render at once in the same cycle. 
2) Apply an inert ngIf to all tabs even if they aren't conditional like `*ngIf="true"` should force tabs to all render in the same cycle.

**Comment by rbirkgit** [comment]
I thought of option 2 above. Didn't try it but should work. Easy enough workaround.

**Comment by rbirkgit** [comment]
Adding an ngIf on the clr-tabs element doesn't seem to fix it, but adding on the remaining tabs themselves seems to work on the stackblitz example above. I will try this in our product.

**Comment by rbirkgit** [comment]
Option 2 above did not work in our case. This because one of our tabs is only enabled depending on the result of an async function. So using *ngIf="true" is not working as it will be enabled too early. May need to write something more elaborate for the dummy *ngIf.

**Comment by gnomeontherun** [comment]
Yeah, the point is you'd need to ensure they all resolve in the same lifecycle, which gets more complicated to demonstrate but works the same in principle.

**Comment by gnomeontherun** [comment]
Another workaround would be to use the CSS version of tabs and manage the aria-owns lifecycle yourself. The more we manage internal state the harder it can be to make things behave as expected in interesting edge cases. 

There is a workaround available for this issue, and we recommend using it for Clarity Angular. We would consider a contribution to address this issue if someone is able to work out a solution. As we build Clarity Core implementations, we expect that this issue won’t be an issue or will be configurable on the application side. To help us clean up our backlog, we are going to close this with a functional workaround available and suggest you follow updates for Clarity Core for enhancements that can support your use case with Clarity Core components.
